kolena/workflow/evaluator.py

```diff
@@ -268,22 +270,25 @@ def _validate_metrics_test_sample_type(metrics_test_sample_type: Type[MetricsTes
 
 
 def _validate_metrics_test_case_type(metrics_test_case_type: Type[DataObject]) -> None:
-    validate_scalar_data_object_type(metrics_test_case_type, supported_list_types=[MetricsTestCase])
+    validate_scalar_data_object_type(
+        metrics_test_case_type,
+        supported_list_types=[MetricsTestCase, ThresholdedMetrics],
```

Comment:
Do we need to validate that `ThresholdedMetrics` is only used as a list type?

In other words, it doesn't make sense to have something like:
```
class ObjectDetectionThresholded(ThresholdedMetrics):
    tp: List[BoundingBox]
    fp: List[BoundingBox]
    fn: List[BoundingBox]
    precision: float
    recall: float

# makes sense
class MyMetrics(MetricsTestSample):
    car_detections: List[ObjectDetectionThresholded]

# doesn't make sense
class MyMetrics2(MetricsTestSample):
    car_detections: ObjectDetectionThresholded
```